lib/membrane_mp4/muxer/cmaf.ex

```diff
@@ -185,51 +220,173 @@ defmodule Membrane.MP4.Muxer.CMAF do
     |> put_in([:sample_queues, pad], %SamplesQueue{
       duration_range: state.chunk_duration_range || DurationRange.new(state.segment_min_duration)
     })
+    |> Map.update!(:pads_registration_order, &[pad | &1])
```

Comment:
Actually, you could place the pad in the proper input group here and get rid of `pads_registration_order` and `prepare_input_groups`. Generating track ids would also be simpler

Reply:
The problem is that we know the input groups only after all output pads are present (only then we know which input pads should be put into an input group together).